sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/core/MultiTenantDBCosmosFactory.java

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.data.cosmos.core;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.spring.data.cosmos.CosmosFactory;
+import com.azure.spring.data.cosmos.config.CosmosConfig;
+import com.azure.spring.data.cosmos.core.convert.MappingCosmosConverter;
+import org.springframework.data.auditing.IsNewAwareAuditingHandler;
+
+/**
+ * Example for extending CosmosFactory for Mutli-Tenancy at the database level
+ */
+public class MultiTenantDBCosmosFactory extends CosmosFactory {
+
+    public String databaseName;
```

Comment:
@kushagraThapar I am just using it here so that I can write to multiple databases, it is not related to the one in CosmosFactory so I will give it a new name so that it is not confusing. In a real example the user will pull this from somewhere like the request in getDatabaseName(), but for this test I am just doing a simple use case where I can set the database name I want to use directly and then having getDatabaseName() return what I set.